modules/api/php/endpoints/candidate/visit/dicom/processes.class.inc

```diff
@@ -0,0 +1,255 @@
+            $handler
+        );
+    }
+
+    /**
+     * Create an array representation of this endpoint's response body
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request): ResponseInterface
+    {
+        if (isset($this->_cache)) {
+            return $this->_cache;
+        }
+
+        // *************************
+        // *** Permission checks ***
+        // *************************
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('imaging_uploader')) {
+            return new \LORIS\Http\Response\JSON\Forbidden();
+        }
+
+        try {
+            $mriuploads = $this->_visit->getMRIUploadsByFilename(
+                $user,
+                $this->_tarname
+            );
+        } catch (\NotFound $e) {
+            return new \LORIS\Http\Response\JSON\NotFound();
+        }
+
+        $array = (new \LORIS\api\Views\Visit\Dicom\Processes($mriuploads))
+            ->toArray();
+
+        $this->_cache = new \LORIS\Http\Response\JsonResponse($array);
+
+        return $this->_cache;
+    }
+
+    /**
+     * Launch the MRI pipeline on the DICOM study.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePOST(ServerRequestInterface $request): ResponseInterface
+    {
+        // *************************
+        // *** Permission checks ***
+        // *************************
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('imaging_uploader')) {
+            return new \LORIS\Http\Response\JSON\Forbidden();
+        }
+
+        $data        = json_decode($request->getBody()->__toString(), true);
+        $processtype = $data['ProcessType'] ?? null;
+        if (!in_array($processtype, ['mri_upload'])) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Invalid ProcessType.'
+            );
+        }
+
+        $mriuploadid = $data['MriUploadID'] ?? null;
+        if (!is_numeric($mriuploadid)) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'Invalid ProcessType.'
+            );
+        }
+
+        try {
+            $mriupload = $this->_visit->getMRIUploadsById(
+                $user,
+                (int) $mriuploadid
+            );
+        } catch (\NotFound $e) {
+            return new \LORIS\Http\Response\JSON\NotFound(
+                'Filename and MRIUploadID do not match'
+            );
+        }
+
+        try {
+            $processid = (new ServerProcessLauncher())->mriupload(
+                $mriuploadid,
+                $mriupload->getUploadLocation()
+            );
+        } catch (\LorisException $e) {
+            error_log($e->getMessage());
```

Comment:
This should be using the PSR-3 logger for exceptions so that the config setting is honoured.

Reply:
https://github.com/aces/Loris/pull/9154/commits/6f3651d8fa430ec4d3ad51242995db097922e62e